go/private/rules/transition.bzl

```diff
@@ -54,6 +64,14 @@ def filter_transition_label(label):
     else:
         return str(Label(label))
 
+def _original_value_setting(setting):
+    if not "//go/config:" in setting:
+        return None
+    name = setting.split(":")[1]
+    return filter_transition_label("@io_bazel_rules_go//go/private/rules:original_" + name)
```

Comment:
It converts `@io_bazel_rules_go//go/config:race` to `//go/config:race` when `io_bazel_rules_go` is the main repo and returns it unchanged otherwise. It may not be necessary here since the "original_" settings are never reference on the command-line, but using it for some settings and not others seemed even more confusing to me - it would also affect the way transition inputs/outputs are declared below.

The bug is fixed, but the full fix is only available in Bazel 5+. Once the minimum version has been updated to 5.0.0, `filter_transition_label` can be dropped.